### ISSUE TITLE
inconsistencies in TM  transport function term names

### ISSUE THREAD

**Issue by ValWood** [geneontology/go-ontology#12587]
oligopeptide transporting ATPase activity
does not return in top hits
oligopeptide-transporting ATPase activity

There is a lot if inconsistency as to how these are named, and I’m not sure which hyphenation is correct but I see:

hyphen after substrate
GO:0015446	arsenite-transmembrane transporting ATPase activity	
GO:0005388	calcium-transporting ATPase activity	
		protein-transmembrane transporting ATPase activity	

hyphen after transmembrane
GO:0015414 nitrate transmembrane-transporting ATPase activity
	   sulfate transmembrane-transporting ATPase activity

no hyphen
GO:0033225	2-aminoethylphosphonate transporting ATPase activity
		oligogalacturonide transporting ATPase activity
GO:0035673	oligopeptide transmembrane transporter activity

etc

could these all  be standardised so we know what to search on (no hyphen seems correct to me?)

**Comment by paolaroncaglia** [comment]
Well this is an interesting subject for a non-native speaker. Personally I tend to put hyphens more often than not (as in non-native ;-)), but Wikipedia seems to agree with @ValWood , so I’ll remove the hyphens from those terms.

https://en.wikipedia.org/wiki/Hyphen#Use_in_English

“The English language does not have definitive hyphenation rules … 
The use of the hyphen in English compound nouns and verbs has, in general, been steadily declining. Compounds that might once have been hyphenated are increasingly left with spaces or are combined into one word. In 2007, the sixth edition of the Shorter Oxford English Dictionary removed the hyphens from 16,000 entries, such as fig-leaf (now fig leaf), pot-belly (now pot belly) and pigeon-hole (now pigeonhole).The increasing prevalence of computer technology and the advent of the Internet have given rise to a subset of common nouns that might have been hyphenated in the past (e.g. "toolbar", "hyperlink", "pastebin").”

**Comment by ValWood** [comment]
well there are some places where it is used  correctly like 
x-dependent
actin-like
I don't know what the rule is but I'm sure there must be one...
I bet @mah11 knows!

**Comment by paolaroncaglia** [comment]
Hi @ValWood,

On second thought…
We discussed this in the office with @dosumis and @tonysawfordebi , and think that this may rather be considered an improvement plan for the search tool at PomBase. I.e., optimally a tool *should* return ‘oligopeptide-transporting ATPase activity’ if you search for ‘oligopeptide transporting ATPase activity’.
We realise this may not be a quick fix, but it should be possible on at least some levels of implementation (e.g., in OBO-Edit if you type ‘oligopeptide transporting ATPase activity’ the autocomplete does suggest ‘oligopeptide-transporting ATPase activity’, though it doesn’t actually return it as a result unless you click on it). 
Plus, many GO term labels do not use hyphens consistently, not just in the transport branch, and it would take a long time to fix them manually. Many have synonyms containing (or not containing) hyphens though. 

Bottom line, I’m going to close this ticket, and I suggest you look into adding a feature in the PomBase search tool instead.   

Thanks!

**Comment by ValWood** [comment]
I think the hyphenation should be grammatically correct in the term name.
Midori always  insisted on it....I do't think this should change.

Also, I was using QuickGO.....

**Comment by ValWood** [comment]
so correct (unhyphenated)
oligopeptide transporting ATPase activity
does not return GO:0015421 in visible hit-list

**Comment by ValWood** [comment]
maybe it is ciorrect to use
"-transporting" (i.e before the verb)?

I usually search on no hyphen because I know "substrate transporter" does not generally have one.
ATPase coupled xxx transmembrane transporter activity
would make them easier to locate, and be clearer primary term names in the ATPase branch...

**Comment by mah11** [comment]
> I think the hyphenation should be grammatically correct (or at least internally consistent) in the primary term name.

I agree. It's not at all just about what any search engine can find; it's primarily about how the terms look in displays, including MOD gene pages, enrichment analysis output, etc. etc as well as search results.

> so correct (unhyphenated)
> oligopeptide transporting ATPase activity

Actually, I would say hyphenated is correct for this, i.e. "oligopeptide-transporting ATPase activity" (Wikipedia's claims notwithstanding).

I have a very strong preference for hyphenating compound adjectives for consistency with the cases where it's essential (there are some situations where hyphenated and unhyphenated versions of a phrase have different meanings, although I don't know if any GO term names are among them).

It gets messier when we have to include "transmembrane" in the mix. It's clear that the best pattern is "x-transporting ATPase" when x is just the name of the substance transported, but "transmembrane" doesn't fit in gracefully. Both "arsenite-transmembrane transporting ATPase activity" and "nitrate transmembrane-transporting ATPase activity" look odd; it wouldn't be any worse to move "transmembrane" to yield "transmembrane nitrate-transporting ATPase activity".

Although I'm really not sure what the best solution is for the transmembrane terms, I don't think that should stop us making the simpler "x-transporting ATPase" terms mutually consistent.

**Comment by mah11** [comment]
> ATPase coupled xxx transmembrane transporter activity

and I would use "ATPase-coupled xxx transmembrane transporter activity" here ...

**Comment by paolaroncaglia** [comment]
@ValWood, @mah11
Summing up:

- Let’s limit to the ATPase activity branch

- Wrt ‘x-transporting ATPase activity’ terms, we resolved that the correct syntax contains the hyphen, i.e. ‘oligopeptide-transporting ATPase activity’ (in fact, most terms in this branch conform to that)

- Searching GO term labels for ‘transporting ATPase activity’ (using OBO-Edit) retrieves 86 matches. All of the ‘non-transmembrane’ labels are already correctly hyphenated, except for the following ones, which I’ll fix:

GO:0033225	2-aminoethylphosphonate transporting ATPase activity
GO:0033154	oligogalacturonide transporting ATPase activity
GO:0015633	zinc transporting ATPase activity

- Within the same branch, the following labels contain the problematic ‘transmembrane’ string:

GO:0102017	alkylphosphonate transmembrane-transporting ATPase activity
GO:0043225	anion transmembrane-transporting ATPase activity
GO:0015446	arsenite-transmembrane transporting ATPase activity
GO:1901242	doxorubicin transmembrane-transporting ATPase activity
GO:1901514	lipo-chitin oligosaccharide transmembrane-transporting ATPase activity
GO:1901243	methionine transmembrane-transporting ATPase activity
GO:0015412	molybdate transmembrane-transporting ATPase activity
GO:0015414	nitrate transmembrane-transporting ATPase activity
GO:0015416	organic phosphonate transmembrane-transporting ATPase activity
GO:0015415	phosphate ion transmembrane-transporting ATPase activity
GO:0015462	protein-transmembrane transporting ATPase activity
GO:0015419	sulfate transmembrane-transporting ATPase activity
GO:0032146	thiosulfate transmembrane-transporting ATPase activity
GO:1901238	tungstate transmembrane-transporting ATPase activity

All these labels follow the pattern ‘x transmembrane-transporting ATPase activity’ except for one (arsenite-transmembrane transporting ATPase activity). I agree that for these cases the current hyphenation is unsatisfactory. I agree with Midori’s proposal to rename them all as ‘ATPase-coupled xxx transmembrane transporter activity’, and retain the original labels as (related) synonyms. 

Please confirm that this is good with you both before I do the edits. Thanks!

**Comment by ValWood** [comment]
That sounds good to me.
Val

**Comment by mah11** [comment]
all sounds fine; thanks

**Comment by paolaroncaglia** [comment]
Fixed 
GO:0033225 2-aminoethylphosphonate transporting ATPase activity
GO:0033154 oligogalacturonide transporting ATPase activity
GO:0015633 zinc transporting ATPase activity

More later

**Comment by paolaroncaglia** [comment]
Also fixed 
GO:0102017 alkylphosphonate transmembrane-transporting ATPase activity
GO:0043225 anion transmembrane-transporting ATPase activity
GO:0015446 arsenite-transmembrane transporting ATPase activity
GO:1901242 doxorubicin transmembrane-transporting ATPase activity
GO:1901514 lipo-chitin oligosaccharide transmembrane-transporting ATPase activity

Not done yet...

**Comment by paolaroncaglia** [comment]
Done.

**Comment by paolaroncaglia** [comment]
Just realised that the TG template "chemical_transmembrane_transport_mf" has a checkbox option for "ATPase activity, coupled to transmembrane movement of substances" that will yield terms named e.g. GO:1901238 tungstate transmembrane-transporting ATPase activity. This is an issue for the TG tracker. Will create ticket there, not sure when it can be addressed though. Will update with link.
